javascript/ql/test/query-tests/Security/CWE-312/passwords.js

```diff
@@ -137,3 +137,19 @@
     console.log(config.y); // NOT OK
     console.log(config[x]); // OK (probably)
 });
+
+function indirectLogCall() {
+	console.log.apply(this, arguments);
+}
+var Util = require('util');
+(function() {
+    var config = {
```

Comment:
Can we add these two test cases as well?

```
console.log(process.env);
console.log(process.env.PATH);
```

Reply:
`console.log(process.env.PATH);` is not marked by the current query. 

Flow-labels should fix that. I'll look into flow-labels.

Reply:
Why would `PATH` contain sensitive data?

Reply:
Esben did you mean `console.log(process.env.PATH);` as an example of something that should _not_ be flagged? I don't think that was clear.

Reply:
`console.log(process.env.PATH)` should not be flagged IMO.

Reply:
You mean that the specific property `Path` should not be flagged?  
But other properties of `process.env` should be flagged?  

For now there is no such whitelist, but I could add such a list for some common benign environment properties.

Reply:
Logging all properties (recursively) on `process.env` through `util.inspect` is dangerous as that may contain confidential data that the programmer **did not anticipate**.

For now, I think explicit properties on `process.env` (like `process.env.PATH`) should not be flagged. 

In between the two extremes, there may be a middle ground where we want to flag stuff like:

```
for (var p in process.env) {
    console.log(process.env[p]); // NOT OK
}
```

but I think that is much less likely to as interesting as `process.env` through `util.inspect`.

I would therefore annotate my examples as:

```
console.log(process.env); // NOT OK
console.log(process.env.PATH); // OK
```